backend/windmill-api/src/slack_approvals.rs

```diff
@@ -0,0 +1,1138 @@
+async fn send_slack_message(
+    client: &Client,
+    bot_token: &str,
+    channel_id: &str,
+    w_id: &str,
+    job_id: Uuid,
+    resource_path: &str,
+    approver: Option<&str>,
+    message: Option<&str>,
+    flow_step_id: &str,
+) -> Result<StatusCode, Box<dyn std::error::Error>> {
+    let url = "https://slack.com/api/chat.postMessage";
+
+    let mut value = serde_json::json!({
+        "w_id": w_id,
+        "job_id": job_id,
+        "path": resource_path,
+        "channel": channel_id,
+        "flow_step_id": flow_step_id,
+    });
+
+    if let Some(approver) = approver {
+        value["approver"] = serde_json::json!(approver);
+    }
+
+    if let Some(message) = message {
+        value["message"] = serde_json::json!(message);
+    }
+
+    let payload = serde_json::json!({
+        "channel": channel_id,
+        "text": "A flow has been suspended. Please approve or reject the flow.",
+        "blocks": [
+            {
+                "type": "section",
+                "text": {
+                    "type": "mrkdwn",
+                    "text": "A flow has been suspended. Please approve or reject the flow."
+                }
+            },
+            {
+                "type": "actions",
+                "elements": [
+                    {
+                        "type": "button",
+                        "text": {
+                            "type": "plain_text",
+                            "text": "View"
+                        },
+                        "action_id": "open_modal",
+                        "value": value.to_string()
+                    }
+                ]
+            }
+        ]
+    });
+
+    tracing::debug!("Payload: {:?}", payload);
+
+    let response = client
+        .post(url)
+        .bearer_auth(bot_token)
+        .json(&payload)
+        .send()
+        .await?;
+
+    if response.status().is_success() {
+        tracing::info!("Interactive Slack approval message sent successfully!");
+        tracing::debug!("Response: {:#?}", response.text().await?);
+    } else {
+        tracing::error!(
+            "Failed to send interactive Slack approval message: {}",
+            response.text().await?
+        );
+    }
+
+    Ok(StatusCode::OK)
+}
+
+async fn get_modal_blocks(
+    db: DB,
+    w_id: &str,
+    job_id: Uuid,
+    resume_id: u32,
+    approver: Option<&str>,
+    message: Option<&str>,
+    trigger_id: &str,
+    flow_step_id: Option<&str>,
+    resource_path: &str,
+    container: Container,
+) -> Result<axum::Json<serde_json::Value>, windmill_common::error::Error> {
+    let res = get_resume_urls_internal(
+        axum::Extension(db.clone()),
+        Path((w_id.to_string(), job_id, resume_id)),
+        Query(QueryApprover { approver: approver.map(|a| a.to_string()) }),
+    )
+    .await?;
+
+    let urls = res.0;
+
+    tracing::debug!("Job ID: {:?}", job_id);
+
+    let (job_kind, script_hash, raw_flow, parent_job_id) = sqlx::query!(
```

Comment:
@uael can you take a look at this new query?

Reply:
I'm not sure to understand the `parent_job` usage, why not directly
```rust
use windmill_common::cache;
let flow_data = cache::job::fetch_flow(&db, job_kind, script_hash).await {
    Ok(data) => data,
    _ => cache::job::fetch_preview_flow(&db, &job_id, raw_flow).await 
};
```
?

Otherwise the rest lgtm!